contracts/vaults/RibbonThetaVault.sol

```diff
@@ -277,13 +277,17 @@ contract RibbonThetaVault is RibbonVault, OptionsThetaVaultStorage {
             lockedBalance
         );
 
-        startAuction();
+        _startAuction();
     }
 
     /**
      * @notice Initiate the gnosis auction.
      */
-    function startAuction() public onlyOwner {
+    function startAuction() external onlyKeeper nonReentrant {
+        _startAuction();
+    }
+
+    function _startAuction() private {
         GnosisAuction.AuctionDetails memory auctionDetails;
 
```

Comment:
not sure if best to disregard this:

At the end of the originally called startAuction function, the optionAuctionID state variable is
modified, violating the recommended Check-Effects-Interaction pattern.